examples/AzureSDKDemoSwift/Source/BlobDownloadViewController.swift

```diff
@@ -0,0 +1,277 @@
+            case let .failure(error):
+                self.showAlert(error: error)
+                self.noMoreData = true
+            }
+        }
+    }
+
+    /// Reload the table view on the UI thread.
+    private func reloadTableView() {
+        DispatchQueue.main.async { [weak self] in
+            self?.tableView.reloadData()
+        }
+    }
+
+    private func reloadTableView(forTransfer transfer: BlobTransfer) {
+        DispatchQueue.main.async { [weak self] in
+            guard let filtered = self?.downloadMap.filter({ _, data in
+                data === transfer
+            }) else { return }
+            let indexes = Array(filtered.keys)
+            self?.tableView.reloadRows(at: indexes, with: .fade)
+        }
+    }
+
+    // MARK: Internal Methods
+
+    internal func playVideo(_ indexPath: IndexPath, _ destination: URL) {
+        downloadMap.removeValue(forKey: indexPath)
+        player.replaceCurrentItem(with: AVPlayerItem(asset: AVAsset(url: destination)))
+        let controller = AVPlayerViewController()
+        controller.player = player
+        present(controller, animated: true) {
+            self.player.play()
+        }
+    }
+
+    // MARK: MSALInteractiveDelegate
+
+    func didCompleteMSALRequest(withResult result: MSALResult) {
+        AppState.account = result.account
+    }
+
+    func parentForWebView() -> UIViewController {
+        hideActivitySpinner()
+        return self
+    }
+}
+
+extension BlobDownloadViewController: UITableViewDelegate, UITableViewDataSource {
+    // MARK: - Table view data source
+
+    func numberOfSections(in _: UITableView) -> Int {
+        return 1
+    }
+
+    func tableView(_: UITableView, numberOfRowsInSection _: Int) -> Int {
+        guard let data = dataSource?.items else { return 0 }
+        return data.count
+    }
+
+    internal func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cellIdentifier = "CustomTableViewCell"
+        guard let blobClient = blobClient,
+            let data = dataSource?.items,
+            let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier, for: indexPath)
+            as? CustomTableViewCell else {
+            fatalError("Preconditions to create CustomTableViewCell not met.")
+        }
+        // configure the cell
+        let blobItem = data[indexPath.row]
+        let blobName = blobItem.name
+
+        cell.keyLabel.text = blobItem.name
+        cell.valueLabel.text = "???"
+        if let blobType = blobItem.properties?.blobType {
+            cell.valueLabel.text = blobType.rawValue
+        }
+        cell.backgroundColor = .white
+
+        // Match any blobs to existing transfers.
+        // Update download map and progress.
+        if let transfer = blobClient.transfer(
+            forBlob: blobName,
+            inContainer: AppConstants.videoContainer,
+            ofType: .download
+        ) {
```

Comment:
Using `transfer(withId)` doesn't work here because the whole point is to see if there is a transfer in progress for a blob on the service. The transfer IDs are generated randomly by transfer manager, so there's nothing to tie them to something on the service.

Reply:
Updated to use your transfers extensions.